mapquest_geocoder.go

```diff
@@ -58,7 +63,11 @@ func (g *MapQuestGeocoder) Request(url string) ([]byte, error) {
 // if one occurs during the geocoding request.
 func (g *MapQuestGeocoder) Geocode(query string) (*Point, error) {
 	url_safe_query := url.QueryEscape(query)
-	data, err := g.Request(fmt.Sprintf("search.php?q=%s&format=json", url_safe_query))
+	key := ""
+	if (MapquestAPIKey != "") {
+		key = "key="+MapquestAPIKey+"&"
+	}
+	data, err := g.Request(fmt.Sprintf("search.php?%sq=%s&format=json",key, url_safe_query ))
```

Comment:
Yup key is totally optional, but for higher quota users is totally needed.  The way I construct the string in line 68 and 105, it  omits the "key=&" in cases where the string is blank. I wasn't sure if explicitly setting the key to "" was a bad thing. I can revisit it if you'd like but the format is basically exactly as you suggest.